### DISCUSSION TITLE
Narrowed return type of children()

### DISCUSSION THREAD

**Discussion by thetarnav** [solidjs/solid#810] [Ideas]
Could the return type of `children()` helper be narrowed to not include `ArrayElement` & `FunctionElement`?
All the arrays and functions are recursively resolved to just Nodes/primitive values, so the types could reflect that. It would help with modifying elements, as there would be less types to worry about.
Something like this could work:
```ts
export type ResolvedElement = Node | (string & {}) | number | boolean | null | undefined
export declare function children(fn: Accessor<JSX.Element>): Accessor<ResolvedElement | ResolvedElement[]>;
```
edit: actually there is `Exclude`:
```ts
export type ResolvedElement = Exclude<JSX.Element, JSX.ArrayElement | JSX.FunctionElement>
```

**Comment by ryansolid** [comment]
They can still be arrays. But you are right, functions get resolved so see some potential here.

**Reply by thetarnav** [reply]
Do you mean it could return a nested array? What's the trick to that? 😅  I'm trying, but it always gets flattened perfectly.

**Reply by ryansolid** [reply]
Oh sorry I meant an array period.. not a nested one.. Gotcha.. you can just have `resolvedElement | resolvedElement[]`
